src/components/app-link/app-link.vue

```diff
@@ -1,11 +1,60 @@
-<script setup>
-  const LinkWithTrailingSlash = defineNuxtLink({
-    trailingSlash: 'append'
-  })
-</script>
-
 <template>
-  <LinkWithTrailingSlash>
-    <slot />
+  <LinkWithTrailingSlash
+    :class="{
+      body: Boolean(variant),
+      'app-link--underline': variant === 'underline',
+    }"
+  >
+    <span
+      v-if="variant === 'underline'"
+      class="app-link__icon"
```

Comment:
Can we name this call `app-link__content` or so? IMO it's not really the icon but actually the content, after which we append the arrow icon.